alacritty/src/config/cursor.rs

```diff
@@ -0,0 +1,149 @@
+use std::cmp;
+
+use serde::Deserialize;
+
+use alacritty_config_derive::{ConfigDeserialize, SerdeReplace};
+use alacritty_terminal::ansi::{CursorShape as VteCursorShape, CursorStyle as VteCursorStyle};
+
+use crate::config::ui_config::Percentage;
+
+const MIN_BLINK_INTERVAL: u64 = 10;
+
+#[derive(ConfigDeserialize, Copy, Clone, Debug, PartialEq)]
+pub struct Cursor {
+    pub style: ConfigCursorStyle,
+    pub vi_mode_style: Option<ConfigCursorStyle>,
+    pub unfocused_hollow: bool,
+
+    thickness: Percentage,
+    blink_interval: u64,
+    blink_timeout: u8,
+}
+
+impl Default for Cursor {
+    fn default() -> Self {
+        Self {
+            thickness: Percentage::new(0.15),
+            unfocused_hollow: true,
+            blink_interval: 750,
+            blink_timeout: 5,
+            style: Default::default(),
+            vi_mode_style: Default::default(),
+        }
+    }
+}
+
+impl Cursor {
+    #[inline]
+    pub fn thickness(self) -> f32 {
+        self.thickness.as_f32()
+    }
+
+    #[inline]
+    pub fn style(self) -> VteCursorStyle {
+        self.style.into()
+    }
+
+    #[inline]
+    pub fn vi_mode_style(self) -> Option<VteCursorStyle> {
+        self.vi_mode_style.map(Into::into)
+    }
+
+    #[inline]
+    pub fn blink_interval(self) -> u64 {
+        cmp::max(self.blink_interval, MIN_BLINK_INTERVAL)
+    }
+
+    #[inline]
+    pub fn blink_timeout(self) -> u64 {
+        const MILLIS_IN_SECOND: u64 = 1000;
+        match self.blink_timeout {
+            0 => 0,
+            blink_timeout => {
+                cmp::max(self.blink_interval * 5 / MILLIS_IN_SECOND, blink_timeout as u64)
```

Comment:
I'd be fine with having the 1000 inlined here since that's pretty self-explanatory. But what is the 5 for?

Reply:
I have no idea what it tries to do.

Reply:
https://github.com/alacritty/alacritty/pull/6172

And it's not like anyone asked on review what it was doing.